### PR TITLE
Update Go version to 1.21

### PR DESCRIPTION
**Pull request by calebbrown**
Bump the version of Go to 1.21.

Includes switching to the stdlib version of "slices"

### CODE REVIEW

File: cmd/collect_signals/Dockerfile [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang@sha256:122f3484f844467ebe0674cf57272e61981770eb0bc7d316d1f0be281a88229f AS base
+FROM golang@sha256:b8e62cf593cdaff36efd90aa3a37de268e6781a2e68c6610940c48f7cdf36984 AS base
```
Comment (maxfisher-g) [inline cmd/collect_signals/Dockerfile:15]: may be worth adding the version specifier too, as well as the hash

File: cmd/criticality_score/Dockerfile [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang@sha256:122f3484f844467ebe0674cf57272e61981770eb0bc7d316d1f0be281a88229f AS base
+FROM golang@sha256:b8e62cf593cdaff36efd90aa3a37de268e6781a2e68c6610940c48f7cdf36984 AS base
```
Comment (maxfisher-g) [inline cmd/criticality_score/Dockerfile:15]: may be worth adding the version specifier too, as well as the hash

File: cmd/csv_transfer/Dockerfile [hunk 3]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang@sha256:122f3484f844467ebe0674cf57272e61981770eb0bc7d316d1f0be281a88229f AS base
+FROM golang@sha256:b8e62cf593cdaff36efd90aa3a37de268e6781a2e68c6610940c48f7cdf36984 AS base
```
Comment (maxfisher-g) [inline cmd/csv_transfer/Dockerfile:15]: may be worth adding the version specifier too, as well as the hash

**Review by maxfisher-g — APPROVED** [review]

File: cmd/collect_signals/Dockerfile (hunk 1 above)
Comment (calebbrown) [reply cmd/collect_signals/Dockerfile]: Done

File: cmd/criticality_score/Dockerfile (hunk 2 above)
Comment (calebbrown) [reply cmd/criticality_score/Dockerfile]: Done

File: cmd/csv_transfer/Dockerfile (hunk 3 above)
Comment (calebbrown) [reply cmd/csv_transfer/Dockerfile]: Done

**Review by maxfisher-g — APPROVED** [review]

**Pull request merged**